src/LuaTransformer.ts

```diff
@@ -40,8 +40,9 @@ interface Scope {
 
 export class LuaTransformer {
     public luaKeywords: Set<string> = new Set([
-        "and", "break", "do",  "else", "elseif", "end",    "false",  "for",  "function", "if",
-        "in",  "local", "nil", "not",  "or",     "repeat", "return", "then", "until",    "while",
+        "and", "break", "do",  "else", "elseif", "end", "false",  "for",    "function",     "if",
+        "in",  "local", "new", "nil", "not",  "or",     "repeat", "return", "self", "then", "until",
```

Comment:
some strange spacing in this array actually

Reply:
agreed (I was just emulating what was there)

Reply:
oh, but I messed it up... 😢